core/templates/components/forms/custom-forms-directives/html-select.component.ts

```diff
@@ -16,33 +16,27 @@
  * @fileoverview Component for the selection dropdown with HTML content.
  */
 
-require('directives/angular-html-bind.directive.ts');
-require('domain/utilities/url-interpolation.service.ts');
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 
-// This component allows user to put html into select's options.
-// 'options' should be an array of objects containing attributes 'id' and 'val'
-// Attribute 'val' is presented to the user. After user selection, the
-// corresponding attribute 'id' is assigned to 'selection'.
+@Component({
+  selector: 'oppia-html-select',
+  templateUrl: './html-select.component.html'
+})
+export class HtmlSelectComponent implements OnInit {
+  @Input() options: { id: string; val: string }[];
+  @Input() selection: number;
```

Comment:
@heyimShivam why are we treating "selection" as a number? It seems like in certain cases (e.g. rule editor in the image region interaction) the "id" and the "val" fields are actually strings. The type conversion in line 39 seems to be resulting in unexpected errors when the "this.selectionAsString" value is not a number. This seems to have resulted in a the regression https://github.com/oppia/oppia/issues/15883.